test/common/tcp_proxy/upstream_test.cc

```diff
@@ -40,17 +42,25 @@ template <typename T> class HttpUpstreamTest : public testing::Test {
   }
 
   void setupUpstream() {
-    config_ = std::make_unique<TunnelingConfigHelperImpl>(config_message_);
+    envoy::config::core::v3::SubstitutionFormatString substitution_format_config;
+    substitution_format_config.mutable_text_format_source()->set_inline_string(
+        config_message_.hostname());
+    Formatter::FormatterPtr hostname_fmt =
+        Formatter::SubstitutionFormatStringUtils::fromProtoConfig(substitution_format_config,
+                                                                  context_);
+    config_ = std::make_unique<TunnelingConfigHelperImpl>(config_message_, std::move(hostname_fmt));
```

Comment:
If you moved formatter construction into helper you wouldn't have to duplicate this logic several times.

Reply:
Done